src/org/openvv/OVVAsset.as

```diff
@@ -438,6 +429,46 @@ package org.openvv {
                 _intervalTimer.addEventListener(TimerEvent.TIMER, onIntervalCheck);
                 _intervalTimer.start();
             }
+    }
+
+        private function setStage(evt:Event = null):void
+        {
+            var ad:DisplayObject = _ad as DisplayObject;
+            if(!ad) return;
+
+            ad.removeEventListener(Event.ADDED_TO_STAGE, setStage);
+            try{
+                _stage = ad.stage;
+            }
+            catch(ignore:Error){
+                //stage is inaccessible
+            }
+            if(!_stage)
+                ad.addEventListener(Event.ADDED_TO_STAGE, setStage);
+        }
+
+        private function getDisplayState(results:OVVCheck):String
+        {
+            var hasStageAccess:Boolean;
+            var displayState:String = StageDisplayState.NORMAL;
+
+            try{
+                displayState   = _stage.displayState;
+                hasStageAccess = true;
+            }
+            catch(ignore:Error){
+                // Either stage was null or we can't access it due to security
+                // restrictions, either way we can ignore this error
+            }
+
+            if(!hasStageAccess && _ad && (_ad is DisplayObject))
+            {
+                if ((_ad.width - (results.objRight - results.objLeft)) > 10 && (_ad.height - (results.objBottom - results.objTop)) > 10) {
```

Comment:
The try-catch part tries to retrieve the display state directly from stage. If stage is null or inaccessible we go to the second method which works as follows:
1. Compare ad dimensions to player dimensions
2. If ad is bigger than player, fullscreen is the only possible explanation, therefore display state should be overridden to fullscreen

On Jan 30, 2015, at 4:58 PM, bdementatbrightroll <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/org/openvv/OVVAsset.as<https://github.com/openvv/openvv/pull/75#discussion_r23875706>:

> +        {
> +            var hasStageAccess:Boolean;
> +            var displayState:String = StageDisplayState.NORMAL;
> +
> +            try{
> +                displayState   = _stage.displayState;
> +                hasStageAccess = true;
> +            }
> +            catch(ignore:Error){
> +                // Either stage was null or we can't access it due to security
> +                // restrictions, either way we can ignore this error
> +            }
> +
> +            if(!hasStageAccess && _ad && (_ad is DisplayObject))
> +            {
> +                if ((_ad.width - (results.objRight - results.objLeft)) > 10 && (_ad.height - (results.objBottom - results.objTop)) > 10) {


Could you explain the logic here?

—
Reply to this email directly or view it on GitHub<https://github.com/openvv/openvv/pull/75/files#r23875706>.